### PR TITLE
Feat: Added Dynamic Schedule

### PR DESCRIPTION
**Pull request by SameerHaq349**
Features added
- schedule api call
- schedule page
- cleaned up formatting for sponsors page

### CODE REVIEW

**Comment by abdullah50053** [comment]
@SameerHaq349 You will need to resolve the merge conflicts before I can approve this PR.

**Comment by nam-t24** [comment]
Potential feature could have mobile view only display 1 day instead of 2 so event cards don’t get squished especially for days with packed events.

**Comment by WillieCubed** [comment]
#### 📝 Changed routes:


| [**/**](https://hackportal-61rsr7qwa-acmutd.vercel.app/) | [**/schedule**](https://hackportal-61rsr7qwa-acmutd.vercel.app/schedule) |
|:-:|:-:|
|<a href="https://hackportal-61rsr7qwa-acmutd.vercel.app/">Screenshot of /</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/puAb60om1TzUa5fvtChVBk1GdGnl2R_rPNauzq-4ug8gwgVLBWvu5vTfxcZmDDqNXt5gAtxVlEP1ZY_2Pyg3fw.png">(view full size)</a> |<a href="https://hackportal-61rsr7qwa-acmutd.vercel.app/schedule">Screenshot of /schedule</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/Sprh0CEJobBTxWxbpYSzCXIq1Nkt8bpaDMUJjF37nS9Z50uPdRiOovt59xXRJWqPkwwZMCkEIEakAzinQiQmEA.png">(view full size)</a> |



| [**/sponsors**](https://hackportal-61rsr7qwa-acmutd.vercel.app/sponsors) |
|:-:|
|<a href="https://hackportal-61rsr7qwa-acmutd.vercel.app/sponsors">Screenshot of /sponsors</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/D5UCZF5cp2Hm-6T4CU15gBbqkKnDLEj2wqZJlXiusHJ06DECVunQ8_1s8kdBA66yv_6cG1g25ZWkZ14lAof46w.png">(view full size)</a> |



And 2 other routes:
- [**/api/schedule**](https://hackportal-61rsr7qwa-acmutd.vercel.app/api/schedule)
- [**/api/sponsor**](https://hackportal-61rsr7qwa-acmutd.vercel.app/api/sponsor)



Commit 8b4dbafb8075daef589a18ce0fab04ae8b28b5be (https://hackportal-61rsr7qwa-acmutd.vercel.app).

File: components/SponsorHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+export default function SponsorHeader(props) {
+  return (
+    <div className="flex flex-col place-items-center md:mr-20 ml-10 mr-10 my-4 ">
+      <h1 className="text-8xl pl-2 decoration-1"> {props.title}</h1>
+      <h2 className="m-2">
+        If you would like to sponsor {props.name}, reachout to us at {props.email}
```
Comment (DedsecKnight) [inline components/SponsorHeader.tsx:8]: I believe `reachout` is a typo.

File: pages/sponsors/index.tsx [hunk 2]
```diff
@@ -1,19 +1,55 @@
...
+
   return (
-    <div className="flex flex-col flex-grow">
+    <div className="flex flex-col flex-grow bg-indigo-200">
       <Head>
         <title>HackPortal - Sponsors</title>
         <meta name="description" content="HackPortal's Sponsors Page" />
       </Head>
-      <section id="subheader" className="p-4"></section>
+      <SponsorHeader
```
Comment (abdullah50053) [inline pages/sponsors/index.tsx:27]: Any way to make this dynamic?

Comment (DedsecKnight) [reply pages/sponsors/index.tsx:27]: I think we can create a document on backend to store those data, and then fetch them at `getServerSideProps`

Comment (abdullah50053) [reply pages/sponsors/index.tsx:27]: Let's do that in 1.1 - we'll need it moving forward anyways to have the information display sitewide

**Review by abdullah50053 — APPROVED** [review]

File: pages/schedule/index.tsx [hunk 3]
```diff
@@ -1,18 +1,223 @@
...
+            {`Location: ${data.location}`}
+          </div>
+        </div>
+      </Appointments.AppointmentContent>
+    );
+  },
+);
+
+export default function Calendar(props: { scheduleCard: ScheduleEvent[] }) {
+  console.log(props.scheduleCard);
```
Comment (DedsecKnight) [inline pages/schedule/index.tsx:188]: Might wanna remove the `console.log`.

**Review by abdullah50053 — APPROVED** [review]

**Review by DedsecKnight — APPROVED** [review]

**Pull request merged**